### PR TITLE
Randomwalk revision

### PR DESCRIPTION
**Pull request by DanielMao1**
1. change random walk impletation (current: outer loop for  one seed random walk, inner loop for each step; previous:  outer loop for one step, inner loop for one seed )
2. add random walk api for homogenous graph(change metapath(vector) to walk_length(int))
3. revision of the benchmark and test file

### CODE REVIEW

**Review by czkkkkkk — APPROVED** [review]
LGTM. But may need to wait for Renjie's PR to merge.

**Comment by gpzlx1** [comment]
We have refactored the graph. So please carefully resolve the confilcts. If you have any problem, contact we with wechat or others.

File: tests/test_hetero_random_walk.cc [hunk 1]
```diff
@@ -16,6 +16,10 @@ static bool check_if_edge_exist(int64_t src, int64_t dst, int64_t *indptr_arr, i
     }
     int64_t begin = indptr_arr[src];
     int64_t end = indptr_arr[src + 1];
+    std::cout << "begin:" << begin << std::endl;
```
Comment (czkkkkkk) [inline tests/test_hetero_random_walk.cc:19]: Remove all std::cout and unnecessary comments.

File: src/cuda/graph_ops.cu [hunk 2]
```diff
@@ -109,6 +111,64 @@ torch::Tensor GraphNormalize(torch::Tensor indptr,
   return out_data;
 }
 
+template <int BLOCK_SIZE>
```
Comment (czkkkkkk) [inline src/cuda/graph_ops.cu:114]: Move these kernels to `src/cuda/random_walk.cu`.

**Review by czkkkkkk — APPROVED** [review]
LGTM except some minor issues.

File: src/cuda/graph_ops.cu (hunk 2 above)
Comment (DanielMao1) [reply src/cuda/graph_ops.cu:114]: OK

**Review by gpzlx1 — APPROVED** [review]
LGTM. But are the minor issues solved?

**Comment by DanielMao1** [comment]
> minor

ongoing

File: tests/test_hetero_random_walk.cc (hunk 1 above)
Comment (DanielMao1) [reply tests/test_hetero_random_walk.cc:19]: OK, unnecessary comments are removed

**Comment by DanielMao1** [comment]
> LGTM. But are the minor issues solved?

yes, already done

File: tests/test_hetero_random_walk.cc (hunk 1 above)
Comment (czkkkkkk) [inline tests/test_hetero_random_walk.cc:19]: std::cout are still there.

Comment (DanielMao1) [reply tests/test_hetero_random_walk.cc]: Sorry for missing that. Already removed it.

**Pull request merged**